### DISCUSSION TITLE
how to type check fn parameters?

### DISCUSSION THREAD

**Discussion by dcsan** [prisma/prisma#11737] [Q&A] [answered]
I have a function that expects a prisma model / instance 

const { collection } = new PrismaClient()

or `prisma.collection` (collection is the name of the type, hope it doesn't collide with anything, i'm using PG as backend)

```ts
import { Prisma, PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

export class Buyer {

    async findColl() {
        const coll = await prisma.nftCollection.findFirst()
        await this.buyItem(coll)
    }

    // this is the param i want to type
    async buyItem(coll: any) {
        clog.info('todo - buy', coll)
    }
}

```

so the param is an instance of `prisma.nftCollection` - I think?
but the lowercase prisma instance isn't defined at compile time

Prisma.NftCol

![image](https://user-images.githubusercontent.com/514002/153394334-7c9aa108-b623-4fa8-abfe-016b1541b8f4.png)

![image](https://user-images.githubusercontent.com/514002/153394675-77a878f7-1143-4e0e-8541-f342bb1904d0.png)

![image](https://user-images.githubusercontent.com/514002/153394716-1d1c5ab2-0835-4893-a96c-1771efc3e15d.png)

so a little puzzled how to type arguments. No point using prisma to just `any` my way through everything...

**Comment by dcsan** [comment, marked as answer]
found answer here
https://stackoverflow.com/questions/71100950/how-to-type-prisma-objects-in-parameters

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
